### PR TITLE
Avoid unnecessary pattern matching against Option and Result

### PR DESCRIPTION
**Pull request by ljedrz**

### CODE REVIEW

**Comment by ljedrz** [comment]
Rebased after https://github.com/rust-lang/rust/pull/52764/commits/4c1cb7534b950d8135279f1ae3aae09137ef7815.

File: src/librustc_typeck/check/method/probe.rs [hunk 1]
```diff
@@ -758,9 +758,8 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
                         self.span, infer::FnCall, &fty);
 
                     if let Some(self_ty) = self_ty {
-                        if let Err(_) = self.at(&ObligationCause::dummy(), self.param_env)
-                            .sup(fty.inputs()[0], self_ty)
-                        {
+                        if self.at(&ObligationCause::dummy(), self.param_env)
+                               .sup(fty.inputs()[0], self_ty).is_err() {
```
Comment (nikomatsakis) [inline src/librustc_typeck/check/method/probe.rs]: Nit: I'd prefer to see `is_err` on its own line here

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
Seems fine apart from the one case, where the `is_err` is too easily overlooked

File: src/librustc_typeck/check/method/probe.rs (hunk 1 above)
Comment (ljedrz) [reply src/librustc_typeck/check/method/probe.rs]: Sounds good, I'll update the commit soon.

**Comment by ljedrz** [comment]
@nikomatsakis done.

**Pull request merged**